src/visitor.rs

```diff
@@ -47,6 +49,32 @@ fn is_extern_crate(item: &ast::Item) -> bool {
     }
 }
 
+/// Creates a string slice corresponding to the specified span.
+pub struct SnippetProvider {
+    /// A pointer to the content of the file we are formatting.
+    big_snippet: *const Rc<String>,
+    /// A position of the start of `big_snippet`, used as an offset.
+    start_pos: usize,
+}
+
+impl SnippetProvider {
+    pub fn span_to_snippet(&self, span: Span) -> Option<&str> {
```

Comment:
Could we use an API such as `with_snippet_for_span` which takes a closure `Fn(&str) -> T` thus making a safe interface here?